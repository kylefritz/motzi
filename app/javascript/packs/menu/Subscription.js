import React, { useState } from "react";
import BuyCredits from "../buy/App";

export const humanizeBreadsPerWeek = (perWeek) => {
  if (perWeek === 0.5) {
    return "Every other week";
  }
  if (perWeek === 1.0) {
    return "Every week";
  }
  if (perWeek === 2.0) {
    return "Two breads per week";
  }
  if (perWeek === 3.0) {
    return "Three breads per week";
  }
  return `${perWeek} breads per week`;
};

export default function Subscription({ user, onRefreshUser, deadlineDay }) {
  const [showBuy, setShowBuy] = useState(false);

  return (
    <>
      <div className="row mt-5">
        <div className="col">
          <h5 className="text-center">Subscriber</h5>
          <div className="text-center mb-2">
            <div className="subscriber-info">{user.name}</div>
            <div>
              <small className="ml-2">
                <a href="/signout" className="text-nowrap">
                  Not you?
                </a>
              </small>
            </div>
          </div>
        </div>
        <div className="col">
          <h5 className="text-center">Frequency</h5>
          <p className="text-center subscriber-info" title={user.breadsPerWeek}>
            {humanizeBreadsPerWeek(user.breadsPerWeek)}
          </p>
        </div>
        <div className="col">
          <h5 className="text-center">Order By</h5>
          <p className="text-center subscriber-info">{deadlineDay} Midnight</p>
        </div>
        <div className="col">
          <h5 className="text-center">Credits</h5>
          <p className="text-center">
            <div className="subscriber-info">{user.credits}</div>
            {onRefreshUser && (
              <button
                type="button"
                className="btn btn-sm btn-link text-nowrap"
                style={{ fontSize: "80%" }}
                onClick={() => setShowBuy(!showBuy)}
              >
                Buy more
              </button>
            )}
          </p>
        </div>
      </div>
      {showBuy && <BuyCredits onComplete={onRefreshUser} user={user} />}
    </>
  );
}
