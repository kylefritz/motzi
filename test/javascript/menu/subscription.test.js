import React from "react";
require("../configure_enzyme");
import { shallow } from "enzyme";

import Subscription, { humanizeBreadsPerWeek } from "menu/Subscription";

test("snapshot", () => {
  const user = {
    name: "kyle",
    credits: 6,
    breadsPerWeek: 1.0,
    sendWeeklyEmail: true,
  };
  const wrapper = shallow(<Subscription user={user} deadlineDay="Sunday" />);
  expect(wrapper).toMatchSnapshot();
});

test("humanizeBreadsPerWeek", () => {
  expect(humanizeBreadsPerWeek(0.5)).toEqual("Every other week");
  expect(humanizeBreadsPerWeek(1.0)).toEqual("Every week");
  expect(humanizeBreadsPerWeek(2.0)).toEqual("Two breads per week");
  expect(humanizeBreadsPerWeek(3.0)).toEqual("Three breads per week");
  expect(humanizeBreadsPerWeek(5.3)).toEqual("5.3 breads per week");
});
