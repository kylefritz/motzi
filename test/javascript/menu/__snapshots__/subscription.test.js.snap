// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h5
        className="text-center"
      >
        Subscriber
      </h5>
      <div
        className="text-center mb-2"
      >
        <div
          className="subscriber-info"
        >
          kyle
        </div>
        <div>
          <small
            className="ml-2"
          >
            <a
              className="text-nowrap"
              href="/signout"
            >
              Not you?
            </a>
          </small>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <h5
        className="text-center"
      >
        Frequency
      </h5>
      <p
        className="text-center subscriber-info"
        title={1}
      >
        Every week
      </p>
    </div>
    <div
      className="col"
    >
      <h5
        className="text-center"
      >
        Order By
      </h5>
      <p
        className="text-center subscriber-info"
      >
        Sunday
         Midnight
      </p>
    </div>
    <div
      className="col"
    >
      <h5
        className="text-center"
      >
        Credits
      </h5>
      <p
        className="text-center"
      >
        <div
          className="subscriber-info"
        >
          6
        </div>
      </p>
    </div>
  </div>
</Fragment>
`;
